extensions/wikia/PhalanxII/PhalanxStatsSpecialController.class.php

```diff
@@ -12,45 +12,118 @@ function __construct( ) {
 	}
 
 	public function index() {
-		wfProfileIn( __METHOD__ );
 		$this->wg->Out->setPageTitle( wfMsg( 'phalanx-stats-title' ) );
 		$this->wg->Out->addBacklinkSubtitle( $this->phalanxTitle );
 
 		if ( !$this->userCanExecute( $this->wg->User ) ) {
-			wfProfileOut( __METHOD__ );
 			$this->displayRestrictionError();
-			return;
+			return false;
 		}
 
 		$par = $this->getPar();
+
+		// Look for paths with a wiki ID like /wiki/Special:PhalanxStats/wiki/125951
 		if ( strpos( $par, 'wiki' ) === 0 ) {
 			// show per-wiki stats
-			list ( , $wikiId ) = explode( "/", $par, 2 );
-			$this->blockWikia( $wikiId );
-		} else {
-			$blockId = $this->wg->Request->getInt( 'blockId', intval( $par ) );
-			if ( !empty( $blockId ) ) {
-				// show block stats
-				$this->blockStats( $blockId );
-			} else {
-				// show help page
-				$this->forward( 'PhalanxStatsSpecial', 'help' );
-			}
+			return $this->forward( 'PhalanxStatsSpecial', 'blockWiki' );
+		}
+
+		// Look for paths with a block ID like /wiki/Special:PhalanxStats/123456
+		$blockId = $this->wg->Request->getInt( 'blockId', intval( $par ) );
+		if ( empty( $blockId ) ) {
+			// show help page
+			return $this->forward( 'PhalanxStatsSpecial', 'help' );
+		}
+
+		// show block stats
+		return $this->blockStats( $blockId );
+	}
+
+	/**
+	 * Show a list of blocks for a specific wiki
+	 *
+	 * @return bool
+	 */
+	public function blockWiki() {
+		$par = $this->getPar();
+		list ( , $wikiId ) = explode( '/', $par, 2 );
+
+		$wikiData = $this->getWikiData( $wikiId );
+		if ( empty( $wikiData ) ) {
+			return false;
 		}
 
-		wfProfileOut( __METHOD__ );
+		// We have a valid id, change title to use it
+		$this->wg->Out->setPageTitle(
+			wfMessage( 'phalanx-stats-title' )->text() . ': ' . $wikiData['url']
+		);
+		$this->wg->Out->addBacklinkSubtitle( $this->title );
+
+		$this->getResponse()->setValues( [
+			'wikiData' => $wikiData,
+			'statsPager' => $this->buildWikiPager( $wikiId ),
+		] );
+
+		return true;
 	}
 
+	/**
+	 * Show information for a specific block ID
+	 *
+	 * @param $blockId
+	 * @return bool
+	 */
 	private function blockStats( $blockId ) {
 		$out = $this->getOutput();
 		$out->setPageTitle( $this->msg( 'phalanx-stats-title' )->text() . "#$blockId" );
 		$out->addBacklinkSubtitle( $this->title );
 
-		$data = Phalanx::newFromId( $blockId );
-
-		if ( !isset( $data["id"] ) ) {
+		$data = $this->getPhalanxData( $blockId );
+		if ( empty( $data ) ) {
 			$out->addWikiMsg( 'phalanx-stats-block-notfound', $blockId );
-			return false; // skip rendering
+			return false;
+		}
+
+		// pull these out of the array, so they don't get used in the top rows
+		$row = $data->toArray();
+		unset( $row['text'], $row['reason'], $row['comment'], $row['ip_hex'] );
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "reason" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "comment" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)